tests/conftest.py

```diff
@@ -281,6 +267,30 @@ def _fc(request, fc_module, new_user=False):
     return fc_module
 
 
+def _port(port, busy_ports, ip=_LOCALHOST):
+    import socket
+    from pykern.pkdebug import pkdlog
+
+    p = int(port)
+    while sirepo.const.PORT_MIN <= p <= sirepo.const.PORT_MAX:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            try:
+                assert p not in busy_ports
+                s.bind(
+                    (
+                        ip,
+                        p,
+                    )
+                )
+                busy_ports.append(p)
+                return str(p)
+            except Exception:
+                pkdlog(f"ip={ip} port={p} in use")
```

Comment:
don't log. it's a normal part of the algorithm

Reply:
[739348c](https://github.com/radiasoft/sirepo/pull/4846/commits/739348cec891a109dbd3ad4390a8c9390d92ba7a)